### DISCUSSION TITLE
REST API endpoints definition

### DISCUSSION THREAD

**Discussion by michaelpiron** [project-origin/wallet#168] [General]
### Description

Hi all,
In #5 , a first version of REST API for the wallet has been built and exposed.

Question: how have the endpoint URLs & resources been defined?

- Looking into:  [/src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/ModelsV1.cs](https://github.com/project-origin/wallet/blob/main/src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/ModelsV1.cs) 
-> I see resources like _ClaimedCertificate_ (versus a GranularCertificate), _AggregatedCertificates, AggregatedClaims_ which are new to me.
- Looking into: [/src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/WalletController.cs](https://github.com/project-origin/wallet/blob/main/src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/WalletController.cs) 
-> there are corresponding API endpoints **v1/aggregate-certificates** and **v1/aggregate-claims**

This deviates from our Elia Group API definition, which we shared earlier. I'd like to understand the rationale for the deviations.
Our definition:
[ETT business architecture_v2023-08-22.pdf](https://github.com/project-origin/wallet/files/13393215/ETT.business.architecture_v2023-08-22.pdf)

Maybe also best to have API endpoint definitions documented at a given moment.
Thanks!

**Comment by MartinSchmidt** [comment]
Hi @michaelpiron, 

I have a link here to [swaggerhub](https://app.swaggerhub.com/apis/Project-Origin/WalletSystem/0.6.0) which shows a representation of what is currently available. Swaggerhub is not automatically updated, I just did it by hand, my plans are to find some proper way to continually release API specs.
 
Yes, the API does not follow what you had made until now 1-to-1, this API was mostly spearheaded by the needs of Energinet and by their developers, but I would like to start a further talk about it, either if we should move more in the direction of what Elia already has defined or if we could collaborate on defining a new one. 

The Aggregate APIs was added to more easily be able to answer summarized queries as then user wouldn't need to pull all the individual records if not needed, but are definitely a early version of what it will be in the future.

There are currently not yet any actions available on the rest API, these are to come, as well as wallet management.

**Comment by AndreAlnor** [comment]
Hi @michaelpiron and @MartinSchmidt how about scheduling a virtual meeting to discuss API design? I am working on a hosted registry and wallet service, so every TSO doesn't have to go through the long development process we have done to enable GC issuing. API's should be designed to serve multiple TSO's so let's try to start with finding a common ground between Energinet and Elia needs. Tuesday January 9th could work for me.
